.github/workflows/chainloop_github_release.yml

```diff
@@ -0,0 +1,116 @@
+# Description: This workflow is used to record a release in Chainloop. It is triggered when a release is published on GitHub.
+# It uses the Chainloop CLI to create an attestation for the release assets, source code and additional materials if provided.
+# The attestation is then pushed to the Chainloop service and the attestation link is added to the release notes.
+# Prior to running this workflow, the Chainloop workflow is onboarded if it does not exist.
+on:
+  workflow_call:
+    inputs:
+      project:
+        description: "The project this workflow belongs to. default: repository name"
+        type: string
+      workflow_name:
+        description: "The workflow name. default: parent workflow filename"
+        type: string
+      additional_materials:
+        description: "Additional materials to be attested. Commas separated list of values"
+        type: string
+    secrets:
+      api_token:
+        description: "Reference: https://docs.chainloop.dev/reference/operator/api-tokens#api-tokens"
+        required: true
+      cosign_key:
+        description: "The private key used to sign the attestation"
+        required: true
+      cosign_password:
+        description: "The password for the private key used to sign the attestation"
+        required: true
+
+jobs:
+  onboard_workflow:
+    name: Onboard Chainloop Workflow
+    uses: chainloop-dev/labs/.github/workflows/chainloop_onboard.yml@4173e015dbd5dc2a8802555c268da63d57bbe576
+    if: github.event_name == 'release' && github.event.action == 'published'
+    with:
+      project: ${{ inputs.workflow_project }}
+      workflow_name: ${{ inputs.workflow_name }}
+    # Pass parent workflow secrets to the child workflow
+    secrets: inherit
+
+  release:
+    name: Record release from GitHub
+    runs-on: ubuntu-latest
+    needs: onboard_workflow
+    # Only run this job when a release is published
+    if: github.event_name == 'release' && github.event.action == 'published'
+    env:
+      CHAINLOOP_WORKFLOW_NAME: ${{ needs.onboard_workflow.outputs.workflow_name }}
+      CHAINLOOP_TOKEN: ${{ secrets.api_token }}
+      GH_TOKEN: ${{ github.token }}
+
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Install Chainloop
+        run: |
+          curl -sfL https://docs.chainloop.dev/install.sh | bash -s
+
+      - name: Initialize Attestation
+        run: |
+          chainloop attestation init --workflow-name ${CHAINLOOP_WORKFLOW_NAME}
+
+      - name: Attest all assets
+        run: |
+          tag=$(echo -n ${{github.ref}} | cut -d / -f3)
+          gh release download $tag -D /tmp/github-release || true
```

Comment:
why the `|| true`?

Reply:
Because if there is no assets to download, source code does not count, it raises an exit code 1 that exists the script. By adding || true we ensure we continue adding the rest of materials.

That's what I mentioned yesterday that I even tried capturing the exit code but it didn't work.

Reply:
That makes sense, for the future, for these non-trivial logical cases, it would be great if you add a comment on why. For example, at the beginning I thought this was because in the past we had a more generic workflow that didn't have release guards.

Reply:
Added.